framework/src/main/java/org/apache/drill/test/framework/TestVerifier.java

```diff
@@ -169,16 +169,14 @@ public TestStatus verifyResultSet(String expectedOutput, String actualOutput, bo
         Map.Entry<ColumnList, Integer> entry = iterator.next();
         ColumnList cl = entry.getKey();
         int count = entry.getValue();
-        while (count > 0 && check(expectedMap, cl)) {
-          count--;
-        }
+	count = getUnexpectedCount(expectedMap, cl, count);
```

Comment:
This looks correct to me. But please do some testing cover cases where 1. you have missing rows; 2. rows not expected. Thanks for working on this improvement.

Reply:
Ok chun I tried with 1 file changing deleting some rows adding additional etc, I ll try with multiple as well in diff folders ...

Reply:
Tried with multiple files ,it is working ,but not a full fledged jenkins job

Reply:
After you run framework pre-commit test and no failure, you can merge.
+1.

Reply:
Although there are other factors as well but the performance has improved as u can see with :- 
http://10.10.104.91:8080/job/Test_Framework_Functional_Tests/148/ - n 40
http://10.10.104.91:8080/job/Test_Framework_Functional_Tests/153/ - n 50

as compared to:-
http://10.10.104.91:8080/job/Test_Framework_Functional_Tests/147 - n 40
http://10.10.104.91:8080/job/Test_Framework_Functional_Tests/154/ - n 50

Reply:
Can we pass the map (entry) to the method instead of row (cl) and occurrence (count)